test/test_modules.py

```diff
@@ -853,6 +853,79 @@ def test_errors(self, device, dtype, module_info, training):
             else:
                 raise NotImplementedError(f"Unknown error type {error_input.error_on}")
 
+    @modules([module for module in module_db if not module.is_lazy])
+    @parametrize('swap', [True, False])
+    @parametrize('set_grad', [True, False])
+    def test_to(self, device, dtype, module_info, training, swap, set_grad):
+        module_cls = module_info.module_cls
+        devices = ['cpu']
+        if torch.cuda.is_available():
+            devices += ['cuda']
+        dtypes = module_info.dtypes
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=False, training=training)
+        torch.nn.utils.set_swap_module_params_on_conversion(swap)
+
+        try:
+            for module_input in module_inputs:
+                c_args, c_kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+                fw_args, fw_kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+
+                m = module_cls(*c_args, **c_kwargs)
+
+                def _to(m, set_grad=False):
```

Comment:
Hm this is not really "old-style" the tests in `test_modules` always take in a device and dtype, the first step is usually to move the constructed module to device/dtype using `module.to()`. I am trying to avoid using `module.to()` (whether swapping is enabled or disabled)  since that is the method that this test is intended to test, added a comment